examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -66,55 +67,62 @@ export class TaskList extends DataObject implements ITaskList {
      * collection pattern -- see the contact-collection example for more details on this pattern.
      */
     private readonly tasks = new Map<string, Task>();
+    // savedData stores data retrieved from the external source.
+    private _savedData: SharedMap | undefined;
+    /*
+    * draftData is used for storage of the draft Fluid data. It's used with savedData
+    * to resolve & synchronize the data.
+    * TODO: Update^ when the sync mechanism is appropriately defined.
+    */
+    private _draftData: SharedMap | undefined;
+
+    private get savedData(): SharedMap {
+        if (this._savedData === undefined) {
+            throw new Error("The savedData SharedMap has not yet been initialized.");
+        }
+        return this._savedData;
+    }
+
+    private get draftData(): SharedMap {
+        if (this._draftData === undefined) {
+            throw new Error("The draftData SharedMap has not yet been initialized.");
+        }
+        return this._draftData;
+    }
 
     public readonly addTask = (id: string, name: string, priority: number): void => {
+
         if (this.tasks.get(id) !== undefined) {
             throw new Error("Task already exists");
         }
-        const nameString = SharedString.create(this.runtime);
-        nameString.insertText(0, name);
-        const priorityCell = SharedCell.create(this.runtime) as ISharedCell<number>;
-        priorityCell.set(priority);
+        const savedNameString = SharedString.create(this.runtime);
+        const draftNameString = SharedString.create(this.runtime);
+
+        savedNameString.insertText(0, name);
+        draftNameString.insertText(0, name);
+
+        const savedPriorityCell = SharedCell.create(this.runtime) as ISharedCell<number>;
+        const draftPriorityCell = SharedCell.create(this.runtime) as ISharedCell<number>;
+
+        savedPriorityCell.set(priority);
+        draftPriorityCell.set(priority);
 
         // To add a task, we update the root SharedDirectory.  This way the change is propagated to all collaborators
         // and persisted.  In turn, this will trigger the "valueChanged" event and handleTaskAdded which will update
         // the this.tasks collection.
-        const data: PersistedTask = {
+        const savedDataPT: PersistedTask = {
             id,
-            name: nameString.handle as IFluidHandle<SharedString>,
-            priority: priorityCell.handle as IFluidHandle<ISharedCell<number>>,
+            name: savedNameString.handle as IFluidHandle<SharedString>,
+            priority: savedPriorityCell.handle as IFluidHandle<ISharedCell<number>>,
```

Comment:
Why are we storing handles here?

Reply:
Not to challenging it - just not sure I understand what's happening here!

Reply:
@ghoshkaj This is a great question that I also asked recently. The DDS itself (SharedString/Cell) is not serializable into a JSON since it's a complex object. We use a handle instead which is a more lightweight plain JS version of the SharedCell or SharedString bc that can be serialized and sent over a websocket. Later in the file, we use handle.get, which uses a factory to "rehydrate" the plain JS version back into the complex DDS object for use.